org.adempiere.base/src/org/compiere/util/CacheMgt.java

```diff
@@ -271,7 +271,7 @@ public int resetLocalCache()
 		CacheInterface[] instances = getInstancesAsArray();
 		for (CacheInterface stored : instances)
 		{
-			if (stored != null && stored.size() > 0)
+			if (stored != null && stored instanceof CCache)

```

Comment:
this change is puzzling, why do you wants to reset an empty cache ?

Reply:
> this change is puzzling, why do you wants to reset an empty cache ?

I noticed that the reset method of the CCacheListener was being called when updating a record, but not when doing a full Cache Reset

The reason is that the resetLocalCache(String,int) is comparing "stored instanceof CCache", while the resetLocalCache() is comparing "stored.size() > 0" - so, the CCacheListener is not called at all if the cache is empty, IMO we don't know which logic can be in those reset methods and it must be consistently calling on both cases.

I'm wondering now too if the getElementCount() has also a problem there, because is checking if "stored.size() > 0" but below is using a different method sizeNoExpire() to get the number.

Reply:
If we wants the checking to be more consistent, we can add the size() > 0 check to resetLocalCache(String,int) , not removing the check at resetLocalCache(). The reason we have the instanceof check at resetLocalCache(String,int) is because we have to cast it to CCache inside that if code block.

Reply:
> If we wants the checking to be more consistent, we can add the size() > 0 check to resetLocalCache(String,int) , not removing the check at resetLocalCache(). The reason we have the instanceof check at resetLocalCache(String,int) is because we have to cast it to CCache inside that if code block.

The approach that I'm trying to implement here (using cache listener in other sessions) doesn't work with the "stored.size() > 0" validation because then is not triggered.
That's why I noticed the inconsistency, it was being triggered by a specific record reset, but not by the general process "Cache Reset".

Reply:
> If we wants the checking to be more consistent, we can add the size() > 0 check to resetLocalCache(String,int) , not removing the check at resetLocalCache().

implemented your suggestion in https://github.com/idempiere/idempiere/pull/2194/commits/e0e9a1e84e17d225534a69c2e283c72558db3ba4